table/block_based_table_reader.cc

```diff
@@ -1612,7 +1617,11 @@ BlockIter* BlockBasedTable::NewDataBlockIterator(
   if (input_iter != nullptr) {
     iter = input_iter;
   } else {
-    iter = new BlockIter;
+    if (is_index) {
```

Comment:
In what case we are calling BlockBasedTable::NewDataBlockIterator() for an index block? If we can get ride of those cases, we can get rid of this if/else here.
I can't see why it should happen here, because we don't put Block object of index blocks here, but IndexReader objects, but here we put blocks to block cache.

Reply:
partitioned index blocks